metrics/metrics_test.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewVoidRegistry(t *testing.T) {
+	registry := NewVoidRegistry()
+
+	registry.ReqsCounter().With("some", "value").Add(1)
+	registry.ReqDurationHistogram().With("some", "value").Observe(1)
+	registry.RetriesCounter().With("some", "value").Add(1)
+}
+
+func TestNewMutliRegistry(t *testing.T) {
+	registries := []Registry{newCollectingRetryMetrics(), newCollectingRetryMetrics()}
+	registry := NewMultiRegistry(registries)
+
+	registry.ReqsCounter().With("key", "requests").Add(1)
+	registry.ReqDurationHistogram().With("key", "durations").Observe(2)
+	registry.RetriesCounter().With("key", "retries").Add(3)
+
+	for _, collectingRegistry := range registries {
+		cReqsCounter := collectingRegistry.ReqsCounter().(*collectingCounter)
+		cReqDurationHistogram := collectingRegistry.ReqDurationHistogram().(*collectingHistogram)
+		cRetriesCounter := collectingRegistry.RetriesCounter().(*collectingCounter)
+
+		if cReqsCounter.counterValue != 1 {
+			t.Errorf("Got value %v for ReqsCounter, want %v", cReqsCounter.counterValue, 1)
```

Comment:
Good idea. I also extracted the `1` to a value with proper float64 type, so that I can use `%f` both times.